ADAL/src/request/ADAcquireTokenSilentHandler.m

```diff
@@ -126,6 +127,13 @@ - (void)acquireTokenByRefreshToken:(NSString*)refreshToken
         request_data[OAUTH2_SCOPE] = _requestParams.scope;
     }
     
+
+    NSString* enrollId = [ADEnrollmentGateway enrollmentIDForTokenUserID:(cacheItem ? (cacheItem.userInformation ? cacheItem.userInformation.userId : nil) : nil)
+                                                           requestUserID:_requestParams.identifier ? _requestParams.identifier.userId : nil];
+
```

Comment:
ADAL finds `cacheItem` by `_requestParams.identifier.userId `, which means `cacheItem.userInformation.userId` will always be the same as `_requestParams.identifier.userId`.
So for silent flow here, checking `cacheItem.userInformation.userId` should be enough.

I guess the fallback for silent flow is:
look up by unique account id (`cacheItem.userInformation.homeUserId`) 
-> look up by user id (`cacheItem.userInformation.userId`) 
-> return whatever we have

` cacheItem.userInformation.homeUserId` property is only available in dev branch, therefore we just pass in nil for 2.6.4 branch:
`[ADEnrollmentGateway enrollmentIDForUniqueAccountID:nil userId:cacheItem.userInformation.userId];`

What do you think?

Reply:
I agree, I've made the changes.  For acquireTokenSilent, I've used the userId from _requestParams first in case there is no cacheItem